diff --git a/test/parallel/test-child-process-spawnsync-validation-errors.js b/test/parallel/test-child-process-spawnsync-validation-errors.js
index c12476c042..d0bc2fd460 100644
--- a/test/parallel/test-child-process-spawnsync-validation-errors.js
+++ b/test/parallel/test-child-process-spawnsync-validation-errors.js
@@ -3,11 +3,12 @@ const common = require('../common');
 const assert = require('assert');
 const spawnSync = require('child_process').spawnSync;
 const signals = process.binding('constants').os.signals;
+const rootUser = process.getuid() === 0;
 
 let invalidArgTypeError;
 let invalidArgTypeErrorCount = 62;
 
-if (common.isWindows) {
+if (common.isWindows || rootUser) {
   invalidArgTypeError =
     common.expectsError({ code: 'ERR_INVALID_ARG_TYPE', type: TypeError }, 42);
 } else {
@@ -64,7 +65,7 @@ function fail(option, value, message) {
 if (!common.isWindows) {
   {
     // Validate the uid option
-    if (process.getuid() !== 0) {
+    if (!rootUser) {
       pass('uid', undefined);
       pass('uid', null);
       pass('uid', process.getuid());
