diff --git a/Makefile b/Makefile
index 062d70b..94564e8 100644
--- a/Makefile
+++ b/Makefile
@@ -391,7 +391,8 @@ test/addons/.buildstamp: $(ADDONS_PREREQS) \
 # .buildstamp is out of date and need a rebuild.
 # Just goes to show that recursive make really is harmful...
 # TODO(bnoordhuis) Force rebuild after gyp update.
-build-addons: | $(NODE_EXE) test/addons/.buildstamp
+build-addons: | $(NODE_EXE)
+	$(MAKE) test/addons/.buildstamp
 
 JS_NATIVE_API_BINDING_GYPS := \
 	$(filter-out test/js-native-api/??_*/binding.gyp, \
