---
diff --git a/test/parallel/test-require-resolve.js b/test/parallel/test-require-resolve.js
index 77f07b394f..b897936696 100644
--- a/test/parallel/test-require-resolve.js
+++ b/test/parallel/test-require-resolve.js
@@ -32,7 +32,7 @@ assert.strictEqual(
   require.resolve(fixtures.path('a')).toLowerCase());
 assert.strictEqual(
   fixtures.path('nested-index', 'one', 'index.js').toLowerCase(),
-  require.resolve(fixtures.path('nested-index', 'one').toLowerCase()));
+  require.resolve(fixtures.path('nested-index', 'one')).toLowerCase());
 assert.strictEqual('path', require.resolve('path'));

 console.log('ok');
---
